harness/determined/pytorch/_pytorch_context.py

```diff
@@ -85,6 +94,48 @@ def wrap_model(self, model: torch.nn.Module) -> torch.nn.Module:
         model_id = len(self.models)
         self._main_model.__setattr__(f"model_{model_id}", model)
 
+        if self.experimental._auto_amp:
+            # Ensure the forward pass is wrapped in an autocast context:
+            saved_model = model
```

Comment:
So something wacky was going on with the scopes and bindings - without the saved_model thing there was infinite recursion because model == self as far as the class is concerned. A function would also clean this up nicely though.